CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+<sup>[[link](#flat-map)]</sup>
+
+  ```Ruby
+  # bad
+  all_songs = users.map(&:songs).flatten.uniq
+
+  # good
+  all_songs = users.flat_map(&:songs).uniq
+  ```
+
+* <a name="reverse-each"></a>
+  Prefer `reverse_each` to `reverse.each` because some classes that `include
+  Enumerable` will provide an efficient implementation. Even in the worst case
+  where a class does not provide a specialized implementation, the general
+  implementation inherited from `Enumerable` will be at least as efficient as
+  using `reverse.each`.
+<sup>[[link](#reverse-each)]</sup>
+
+  ```Ruby
+  # bad
+  array.reverse.each { ... }
+
+  # good
+  array.reverse_each { ... }
+  ```
+
+## Naming
+
+* <a name="english-identifiers"></a>
+  Name identifiers in English.
+<sup>[[link](#english-identifiers)]</sup>
+
+  ```Ruby
+  # bad - identifier using non-ascii characters
+  заплата = 1_000
+
+  # bad - identifier is a Bulgarian word, written with Latin letters (instead of Cyrillic)
+  zaplata = 1_000
+
+  # good
+  salary = 1_000
+  ```
+
+* <a name="snake-case-symbols-methods-vars"></a>
+  Use `snake_case` for symbols, methods and variables.
+<sup>[[link](#snake-case-symbols-methods-vars)]</sup>
+
+  ```Ruby
+  # bad
+  :'some symbol'
+  :SomeSymbol
+  :someSymbol
+
+  someVar = 5
+
+  def someMethod
+    ...
+  end
+
+  def SomeMethod
+   ...
+  end
+
+  # good
+  :some_symbol
+
+  def some_method
+    ...
+  end
+  ```
+
+* <a name="camelcase-classes"></a>
+  Use `CamelCase` for classes and modules.  (Keep acronyms like HTTP, RFC, XML
+  uppercase.)
+<sup>[[link](#camelcase-classes)]</sup>
+
+  ```Ruby
+  # bad
+  class Someclass
+    ...
+  end
+
+  class Some_Class
+    ...
+  end
+
+  class SomeXml
+    ...
+  end
+
+  class XmlSomething
+    ...
+  end
+
+  # good
+  class SomeClass
+    ...
+  end
+
+  class SomeXML
+    ...
+  end
+  
+  class XMLSomething
+    ...
+  end
+  ```
+
+* <a name="snake-case-files"></a>
+  Use `snake_case` for naming files, e.g. `hello_world.rb`.
+<sup>[[link](#snake-case-files)]</sup>
+
+* <a name="snake-case-dirs"></a>
+  Use `snake_case` for naming directories, e.g.
+  `lib/hello_world/hello_world.rb`.
+<sup>[[link](#snake-case-dirs)]</sup>
+
+* <a name="one-class-per-file"></a>
+  Aim to have just a single class/module per source file. Name the file name
+  as the class/module, but replacing CamelCase with snake_case.
+<sup>[[link](#one-class-per-file)]</sup>
+
+* <a name="screaming-snake-case"></a>
+  Use `SCREAMING_SNAKE_CASE` for other constants.
+<sup>[[link](#screaming-snake-case)]</sup>
+
+  ```Ruby
+  # bad
+  SomeConst = 5
+
+  # good
+  SOME_CONST = 5
+  ```
+
+* <a name="bool-methods-qmark"></a>
+  The names of predicate methods (methods that return a boolean value) should
+  end in a question mark.  (i.e. `Array#empty?`). Methods that don't return a
+  boolean, shouldn't end in a question mark.
+<sup>[[link](#bool-methods-qmark)]</sup>
+
+* <a name="dangerous-method-bang"></a>
```

Comment:
I like it for stuff like `launch_the_nukes`. For instance, when writing billing code `process_payment!` is a good indication that you better invoke that thing carefully. For a project like logstash, that's less risky, but occasions may still arise.